sway-core/src/language/ty/expression/expression.rs

```diff
@@ -633,8 +633,21 @@ impl TyExpression {
             TyExpressionVariant::Break => {}
             TyExpressionVariant::Continue => {}
             TyExpressionVariant::Reassignment(reass) => {
-                if let TyReassignmentTarget::Deref(expr) = &reass.lhs {
-                    expr.check_deprecated(engines, handler, allow_deprecated);
+                if let TyReassignmentTarget::DerefAccess { exp, indices } = &reass.lhs {
+                    exp.check_deprecated(engines, handler, allow_deprecated);
+                    for indice in indices {
+                        match indice {
+                            ProjectionKind::StructField { name: _ } => {}
```

Comment:
We should probably consider that a field access for deprecation purposes, right?